plugins/bgp/bgp-plugin.go

```diff
@@ -40,12 +63,135 @@ func (p *BgpPlugin) Init() error {
 	p.KVScheduler.RegisterKVDescriptor(gd)
 
 	// register descriptor for bgp peer config
+
 	pd := descriptor.NewPeerConfDescriptor(p.Log, p.BGPServer)
 	p.KVScheduler.RegisterKVDescriptor(pd)
 
+	p.watchCloser = make(chan string)
+	watcher := p.Deps.KVStore.NewWatcher(nodePrefix)
+	err := watcher.Watch(p.onChange, p.watchCloser, "")
+	if err != nil {
+		return err
+
+	}
 	return nil
 }
 func (p *BgpPlugin) Close() error {
 	log.Println("Closing BgpPlugin Application.")
 	return nil
 }
+
+//tutorial says keyval.protowatchresp but i couldnt find it
+//in the  cn-infra/db/keyval/proto_watcher_api.go in our vendor folder there isnt a ProtoWatchResp
+// but on the github for cn infra, there is one
+func (p *BgpPlugin) onChange(resp datasync.ProtoWatchResp) {
+	//key := resp.GetKey()
+
+	//Getting ip
+	value := new(vppnode.VppNode)
+	changeType := resp.GetChangeType()
+	if changeType == datasync.Delete {
+		if prevValExist, err := resp.GetPrevValue(value); err != nil {
+			logging.Errorf("get value error: %v", err)
+			return
+		} else if !prevValExist {
+			logging.Errorf("Delete and Previous Value Does not Exist")
+			return
+		}
+	} else {
+		if err := resp.GetValue(value); err != nil {
+			logging.Errorf("get value error: %v", err)
+			return
+		}
+	}
+	ip := value.IpAddresses[0]
+	ipParts := strings.Split(ip, "/")
+	ip = ipParts[0]
+	//prefixlen, err := strconv.Atoi(ipParts[1])
+
+	client, err := remote.CreateHTTPClient("")
+	if err != nil {
+		logging.Error("create http client error: %v", err)
+		return
+	}
+
+	//Getting Ipam Info
+	b, err := getNodeInfo(client, ip, getIpamDataCmd)
+	if err != nil {
+		logging.Errorf("getnodeinfo error: %v", err)
+		return
+	}
+
+	ipam := ipv4net.IPAMData{}
+	err = json.Unmarshal(b, &ipam)
+	if err != nil {
+		logging.Errorf("unmarshal error: %v", err)
+		return
+	}
+
+	//Setting Route info
+	nlri, _ := ptypes.MarshalAny(&bgp_api.IPAddressPrefix{
+		Prefix:    ipam.PodSubnetThisNode,
+		PrefixLen: 24,
+	})
+
+	a1, _ := ptypes.MarshalAny(&bgp_api.OriginAttribute{
+		Origin: 0,
+	})
+	a2, _ := ptypes.MarshalAny(&bgp_api.NextHopAttribute{
+		NextHop: ip,
+	})
+	attrs := []*any.Any{a1, a2}
+	switch changeType {
+	case datasync.Put:
+		_, err := p.Deps.BGPServer.AddPath(context.Background(), &bgp_api.AddPathRequest{
+			Path: &bgp_api.Path{
+				Family: &bgp_api.Family{Afi: bgp_api.Family_AFI_IP, Safi: bgp_api.Family_SAFI_UNICAST},
+				Nlri:   nlri,
+				Pattrs: attrs,
+			},
+		})
+		if err != nil {
+			logging.Errorf("AddPath: %v", err)
+		}
+	case datasync.Delete:
+		err := p.Deps.BGPServer.DeletePath(context.Background(), &bgp_api.DeletePathRequest{
```

Comment:
p.log.Info the fact that we have a Delete operation, showing which NLRI we're deleting